### ISSUE TITLE
Refactoring: Split off more code from plugin/src/lib.rs into submodules

### ISSUE THREAD

**Issue by llogiq** [llogiq/mutagen#99]
The plugin source is currently split into two files, one of which is auto-generated by `build.rs`. The `lib.rs` is getting unwieldy. Time to browse for sets of functionality that can be put into their own modules, before navigating the source gets too cumbersome.

**Comment by gnieto** [comment]
Do you mean splitting each kind of mutation into a specific `MultiModifier`? If you don't mean this, what would you think about doing that?

**Comment by llogiq** [comment]
No, I'm rather thinking about other random stuff, like our custom TyOcc + Hash + Equals implementation. The default type whitelist stuff is likely to be removed anyway when #100 is done.

**Comment by rexto** [comment]
I'd like to help with this issue. I'm a complete beginner, I just finished reading Rust-book.
